src/app/util/attribute-storage.cpp

```diff
@@ -217,7 +217,25 @@ void emberAfEndpointConfigure()
         emAfEndpoints[ep].parentEndpointId = fixedParentEndpoints[ep];
 
         emAfEndpoints[ep].bitmask.Set(EmberAfEndpointOptions::isEnabled);
-        emAfEndpoints[ep].bitmask.Set(EmberAfEndpointOptions::isFlatComposition);
+        bool isFlatComposition = false;
```

Comment:
Suggest type of `EmberAfEndpointOptions endpointCompositionType = EmberAfEndpointOptions::isTreeComposition` and override it to `isFlatComposition` when you detect the device type. That allows you to do `emAfEndpoints[ep].bitmask.Set(endpointCompositionType)` after the loop, rather than having an if-block.